### ISSUE TITLE
Screenreader only checkbox doesn't work as described

### ISSUE THREAD

**Issue by vossisboss** [wagtail/news-template#43]
The Statistics Block contains a checkbox with help text that says "If checked, the heading will be hidden from view and avaliable to screen-readers only."

When it is text, the title of the section is still visible. See screenshot here:

![Image](https://github.com/user-attachments/assets/54cae3e0-7d29-41b2-8840-d60ddd3c7bc0)

The Statistics Block is the only default block using this feature. The code for the checkbox occurs in `BaseSectionBlock`.

We should probably re-evaluate whether `BaseSectionBlock` is the appropriate location for this feature and consider removing it if we are not prepared to use it in other blocks more broadly.
